integration_tests/test_pass_compare.py

```diff
@@ -0,0 +1,13 @@
+from ltypes import i32
+
+def f():
+    a11: tuple[i32, i32]
+    b11: tuple[i32, i32]
```

Comment:
Test for nested tuples as well.

Reply:
> This case is not yet covered in the tests. Please test for this first.

I avoided using `f32` because of precision issues in comparison. I added a test for the mixed types: 
```py
    a: tuple[tuple[i32, i32], str, bool]
    b: tuple[tuple[i32, i32], str, bool]
    a = (a11, 'ok', True)
    b = (b11, 'ok', True)
    assert a == b
    b = (b11, 'notok', True)
    assert a != b
```